### ISSUE TITLE
[Bug] 防下移仍失效 + 横屏回竖屏回不到原位：resetStockAnchor 在每次 surface detach 清零，锚点无一次继承

### ISSUE THREAD

**Issue by aodianjun** [aodianjun/com.aodianjun.hyperglow.cnplus#36]
## 问题

0.3.94 按 issue #33 修好"控制器更替丢锚 + managed 覆盖 pin"后,**防下移在现场仍失效**;并新增一个可观测的现象:**横屏切回竖屏后,歌词/时钟回不到原位**。

## 现场(0.3.94 / versionCode 121,debug 版)

配置:`aod_clock_follow=false`(锚定)、`suppress_stock_aod_content=false`、`aod_rotate_with_device=true`。

0.3.94 新增的五类日志全部正常输出,但暴露出**锚点被高频清零**:

```
15:48:04.966  Position state rebuilt; anchor seeded from request
15:48:05.002  Stock anchor reset (surface detached)
15:49:00.003  Stock clock pinned anchorY=390.0 appliedY=390.0 requestedY=-5.625 offset=0
15:49:19.570  Stock anchor reset (surface detached)
15:49:23.083  Position state rebuilt; anchor seeded from request
15:49:23.108  Stock anchor reset (surface detached)
15:49:28.713  Stock anchor reset (surface detached)
15:49:37.610  Position state rebuilt; anchor seeded from request
15:49:37.648  Stock anchor reset (surface detached)
15:50:00.012  Stock clock pinned anchorY=260.0 appliedY=260.0 requestedY=-96.75 offset=0
...
15:53:00.010  Stock clock pinned anchorY=390.0 appliedY=390.0 requestedY=-5.625 offset=0
16:17:49.317  Stock anchor reset (surface detached)
16:18:05.626  Position state rebuilt; anchor seeded from request
16:18:05.642  Stock anchor reset (surface detached)
...
```

**统计(单次日志文件):**

| 日志 | 次数 |
|---|---|
| `Stock anchor reset (surface detached)` | **20** |
| `Position state rebuilt; anchor seeded from request` | **10** |
| `Position state rebuilt; anchor inherited y=…` | **0** |
| `Stock clock pinned …` | **4** |

**两个关键点:**

1. **锚点被反复清零**(20 次),且重建时**无一例外是 `seeded from request`** —— 即 `inheritedAnchorY == null`,继承链每次都已断开,锚点只能取"当前(可能已漂移)的请求值"
2. **钉住日志里 anchorY 在 `390.0` / `260.0` 之间跳变** —— 锚点位置本身不稳定,印证"每次重播种到当时位置"

**结果**:钉住虽然在 `pin=true` 时按 `anchorY` 生效(日志中 `appliedY == anchorY != requestedY`),但锚点本身在会话内被反复重置到漂移后的位置 → **对用户表现为"防下移失效"**。

## 根因

**`resetStockAnchor()` 的触发条件(每次 `detachCurrent()`)远宽于其设计假设("会话结束")。**

`AodSurfaceController.detachCurrent()`:

```kotlin
AodPositionHook.setIntegralClockPin(false)
// 会话结束清空跨 controller 继承的时钟锚定(issue #33):
// 下次进入 AOD 重新锚定到当前系统位置。
AodPositionHook.resetStockAnchor()          // ← 每次 detach 都清
clockPinActive = false
AodSurfaceHook.clearSuppressedState()
AodOrientationMonitor.detach()
```

`AodPositionHook`:

```kotlin
@Volatile private var inheritedAnchorY: Float? = null

fun resetStockAnchor() {
    inheritedAnchorX = null
    inheritedAnchorY = null
    HookLogger.i(TAG, "Stock anchor reset (surface detached)")
}

private fun controllerStateFor(controller: Any): ControllerState {
    controllerStates[controller]?.let { return it }
    val seeded = seedControllerState(inheritedAnchorX, inheritedAnchorY)   // inherited 为 null → 用请求值
    controllerStates[controller] = seeded
    HookLogger.i(TAG, "Position state rebuilt; anchor " +
        (if (inheritedAnchorY != null) "inherited y=$inheritedAnchorY" else "seeded from request"))
    return seeded
}
```

**问题在于 `detachCurrent()` 并不等于"会话结束"。** 现场日志显示 surface 在一次 AOD 会话内反复重建(`LinkageTransition Surface ready` 高频、`detachCurrent` 随之触发),于是:

```
detach → inheritedAnchor = null
      → controller 重建 → seeded from request(锚点 = 当前漂移位置)
      → 再 detach → 再清 …
```

**锚点因此几乎从未真正跨重建保持过,`inherited y=…` 一次都没出现。**

### 为什么"横屏回竖屏回不到原位"

同一根因:**旋转会触发 AOD surface 重建** → `detachCurrent()` → 锚点清零 → 在"当前(已漂移/已位移)的位置"重新播种 → 系统时钟/歌词以新位置为锚,自然回不到旋转前的位置。

## 建议

1. **收紧 `resetStockAnchor()` 的触发条件**
   - 只应在**真正跨会话**(AOD 完全退出、`AodPowerCoordinator`/lifetime guard 判定会话结束)时清锚,而不是每次 `detachCurrent()`
   - 可引入会话标识(如 `attachmentGeneration` 之外的"会话代"或 power session 状态),仅在会话代递增时重置
2. **区分"重建"与"会话结束"**
   - `detachCurrent()` 目前同时承担"重建前清理"和"会话结束清理"两种语义,建议拆分
3. **重建时若继承值为 null,不要静默重播种到请求值**
   - 至少保留一条"锚点退化"的显式日志(当前 `seeded from request` 已能体现,但建议同时在 `AodSurfaceController` 侧提示该退化会影响防下移)
4. **旋转导致的重建应保留锚点**
   - 旋转只改变画布逻辑帧,不改变用户对"时钟/歌词竖直位置"的期望,不应重置竖向锚定
5. **回归验证**
   - 复现现场:一次 AOD 会话内应出现 `inherited y=…` 而非清一色 `seeded from request`
   - 统计 `reset (surface detached)` 次数应远小于 `pinned` 次数

## 备注

- 本条是 issue #33 修复引入的**新问题**(锚点继承机制 + 过于宽松的清零点),不是 #33 未修
- 现场为 2026-09-18 15:48-16:20 的 AOD 会话,日志来自 LSPosed modules 日志与 `files/diagnostic-trace.log`
- 相关:issue #33(锚点随 controller 更替丢失)、#26(钉住时钟需求)、#32(横屏裁剪,已修)
